src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1514,6 +1523,21 @@ jint Arguments::set_ergonomics_flags() {
   return JNI_OK;
 }
 
+void Arguments::set_ergonomics_profile() {
+  validate_ergonomics_profile();
+
+  if (FLAG_IS_DEFAULT(ErgonomicsProfile)){
+#ifdef LINUX
+    if (OSContainer::is_containerized()){
+      FLAG_SET_ERGO(ErgonomicsProfile, "dedicated");
+    }
```

Comment:
That's the intention: if the JVM believes it is inside a container, by default it will activate the `dedicated` profile.

Reply:
Right, but `OSContainer::is_containerized()` returns `true` on a physical non-containerized system too on Linux ([JDK-8261242](https://bugs.openjdk.org/browse/JDK-8261242)). Keep that in mind. This can potentially break many physical deployments by using more than it used to for its heap size.

Reply:
I will do some tests regarding is_containerized as well. We knew that JDK-8261242 would potentially block this proposal. We may have to tackle that first.

What tests have you done so far to improve the container discovery?

Reply:
> I will do some tests regarding is_containerized as well. We knew that JDK-8261242 would potentially block this proposal. We may have to tackle that first.

JDK-8261242 is a must fix before "dedicated" can be enabled on the basis of `OSContainer::is_containerized()` as explained elsewhere.

> What tests have you done so far to improve the container discovery?

It's in JDK-8261242. I'll keep that bug updated with my findings.